### PR TITLE
Collection save api

### PR DESCRIPTION
**Pull request by mcmuglim**
@tmillett R 👍
@adzellman O

### CODE REVIEW

File: www/CQMContext.js [hunk 1]
```diff
@@ -311,4 +323,94 @@ CQMContext.prototype._getEntitlementInfo = function(entityNames, successCallback
...
+ *                  The callback is given an array that contains two values:
+ *                  1. The latest known entity that was saved
+ *                  2. The download percentage
+ * @param {Function} errorCallback (OPTIONAL)
+ */
+CQMContext.prototype.saveEntity = function(entity, isSilent, successCallback, errorCallback) {
+
+    var success = successCallback && function(json) {
+        var returnEntity = new Entity(json.rawEntity)
+        successCallback([returnEntity, json.downloadPct]);
```
Comment (tmillett) [inline www/CQMContext.js]: hm so the only way someone can know if the call was successful is to wait for progress updates? So if this method was called and then for some reason it was queued (like another saveEntity call was sent immediately after) the callback may not be called for a while?

Comment (tmillett) [reply www/CQMContext.js]: The workaround is they could call getEntityStatus if they wanted right?

Comment (mcmuglim) [reply www/CQMContext.js]: Good point. We could return something to the successCallback earlier. Like when an operation was created successfully and give it a 0% to start with. 
Or/And return an extra field, like what we returned for getEntityStatus -- ProgressState.

File: www/CQMContext.js [hunk 2]
```diff
@@ -52,6 +52,18 @@ var Entity = function(newEntity) {
 };
 
 /**
+ * This class represents entity download progress state
+ */
+var ProgressState = function(state) {
```
Comment (mttmllns) [inline www/CQMContext.js]: Do we use this?  I don't see `new ProgressState()` called anywhere.

File: www/CQMContext.js [hunk 3]
```diff
@@ -311,4 +323,94 @@ CQMContext.prototype._getEntitlementInfo = function(entityNames, successCallback
     exec(successCallback, fail, "CQMContext", "getEntitlementInfo", [entityNames]);
 };
 
+/**
+ * Get status for an Entity such as isSavable, progressPct, progressState. (Current only supports Collection)
```
Comment (mttmllns) [inline www/CQMContext.js]: progressPct doesn't match up with the native code in https://git.corp.adobe.com/dps/viewer-android/pull/1017 which uses downloadPercent.  The latter seems more descriptive.

File: www/CQMContext.js [hunk 4]
```diff
@@ -311,4 +323,94 @@ CQMContext.prototype._getEntitlementInfo = function(entityNames, successCallback
...
+ *
+ * @param {Entity} or {EntityList} Entity we want to get status for or EntityList we want to get status for.
```
Comment (mttmllns) [inline www/CQMContext.js]: You're missing the actual parameter name on some of the docs in this file (should it be "entityOrEntityList" here?)  "entity" and "isSilent" below, etc.

**Comment by mcmuglim** [comment]
Hi @tmillett @adzellman, this is ready for maybe another quick review pass and would then be mergable. I added argchecks and updated README documentation.

**Pull request merged**